files/bootstrap.sh

```diff
@@ -585,22 +585,14 @@ if command -v nvidia-smi &> /dev/null; then
     sudo nvidia-smi -pm 1 # set persistence mode
     sudo nvidia-smi --auto-boost-default=0
 
-    GPUNAME=$(nvidia-smi -L | head -n1)
-    echo $GPUNAME
-
-    # set application clock to maximum
-    if [[ $GPUNAME == *"A100"* ]]; then
-      nvidia-smi -ac 1215,1410
-    elif [[ $GPUNAME == *"V100"* ]]; then
-      nvidia-smi -ac 877,1530
-    elif [[ $GPUNAME == *"K80"* ]]; then
-      nvidia-smi -ac 2505,875
-    elif [[ $GPUNAME == *"T4"* ]]; then
-      nvidia-smi -ac 5001,1590
-    elif [[ $GPUNAME == *"M60"* ]]; then
-      nvidia-smi -ac 2505,1177
+    MAX_CLOCKS_XML_NO_BREAKS=$(nvidia-smi -q --xml-format | sed -n '/<max_clocks>/, /<\/max_clocks>/p')
+    MAX_MEMORY_CLOCK_MHZ=$(echo $MAX_CLOCKS_XML_NO_BREAKS | sed -e 's/.*<mem_clock>\([0-9]*\) MHz<\/mem_clock>.*/\1/' )
+    MAX_GRAPHICS_CLOCK_MHZ=$(echo $MAX_CLOCKS_XML_NO_BREAKS | sed -e 's/.*<graphics_clock>\([0-9]*\) MHz<\/graphics_clock>.*/\1/' )
+    NUMBER_RE='^[0-9]+$'
```

Comment:
I don't love that this method of parsing is fragile to changes in an XML output format, but I was also hesitant to rely on more sophisticated XML parsing method which relied on extra dependencies or more complicated logic.

As is, this logic parses the data provided by the current versions of `nvidia-smi`, which looks like this...

```
...
		</applications_clocks>
		<default_applications_clocks>
			<graphics_clock>1710 MHz</graphics_clock>
			<mem_clock>6251 MHz</mem_clock>
		</default_applications_clocks>
		<max_clocks>
			<graphics_clock>1710 MHz</graphics_clock>
			<sm_clock>1710 MHz</sm_clock>
			<mem_clock>6251 MHz</mem_clock>
			<video_clock>1500 MHz</video_clock>
		</max_clocks>
		<max_customer_boost_clocks>
			<graphics_clock>1710 MHz</graphics_clock>
		</max_customer_boost_clocks>
...
```

It makes the following assumptions:

* All GPUs attached to an instance support the same application clock speeds as the first GPU
* No XML properties or whitespace will be added to the `max_clocks`, `mem_clock`, or `graphics_clock` XML nodes

I see value in simplicity here, but also recognize these limitations may be less than ideal.